porcupaine/numerical_model/process_eval_new_numerical_data.py

```diff
@@ -1,99 +1,106 @@
-
-    data = {
-    'project_category': [project_category],
-    'district': [district],
-    'budget': [budget]
-    }
-
-    # Create a DataFrame from the dictionary
-    df = pd.DataFrame(data)
-
-    # Save the DataFrame to a CSV file
-    df.to_csv('data/demo_project.csv', index=False)
-
-    predict_project_success("data/demo_project.csv", "numerical_logistic_regression_model.pkl")
-
-
-# Example usage:
-if __name__ == "__main__":
-    # Example file path for the new project data
-    # file_path = "data/new_project.csv"
-
-    # Path to the saved logistic regression model
-    # model_path = 'numerical_logistic_regression_model.pkl'
-
-    # Call the function and get the predicted success chance
-    # success_chance = predict_project_success(file_path, model_path)
-
-    # print(f"The predicted chance of success for the new project is: {success_chance:.2f}")
-
-    print(demo_predict_project_success("Senioři", "Brno - Bohunice", 2500000))
+import pandas as pd
+import joblib
+from porcupaine.settings import *
+from porcupaine.preprocessing.preprocess_numerical_data import preprocess_data
+
+
+# Helper function: One-Hot Encoding with Alignment
+def one_hot_encode_with_alignment(data, columns, training_columns):
+    """
+    Performs one-hot encoding on specified categorical columns and aligns the resulting columns
+    with the columns used during model training.
+
+    Args:
+    - data (pd.DataFrame): The input DataFrame to encode.
+    - columns (list): A list of column names to perform one-hot encoding on.
+    - training_columns (list): The columns that were used during model training.
+
+    Returns:
+    - pd.DataFrame: The DataFrame with one-hot encoded columns, aligned with training columns.
+    """
+    # Apply one-hot encoding
+    data = pd.get_dummies(data, columns=columns, dtype=int)
+
+    # Reindex the columns to match the training columns (add missing columns as 0)
+    data = data.reindex(columns=training_columns, fill_value=0)
+    return data
+
+
+# Helper function: Transform and Scale the "budget" column
+def transform_and_scale_budget(value):
+    """
+    Transforms the 'budget' column by rounding it to the nearest million and scaling it
+    according to predefined scaling values.
+
+    Args:
+    - value (float): The budget value to transform.
+
+    Returns:
+    - float: The scaled budget value.
+    """
+    if value < 1000000:
+        rounded_value = 1000000
+    else:
+        rounded_value = round(value, -6)  # Round to the nearest million
+
+    scaling_map = {
+        1000000: 0,
+        2000000: 0.25,
+        3000000: 0.50,
+        4000000: 0.75,
+        5000000: 1
+    }
+    return scaling_map.get(rounded_value, 1)
+
+
+# Main function: Predict Project Success
+def predict_project_success(dataframe: pd.DataFrame, model_path):
+    """
+    Preprocesses the training data to get the columns, performs one-hot encoding for the new categorical columns and

```

Comment:
this function is quite general supporting any kinds of models which have `predict_proba`. Even more robust function would be checking existence of `predict_proba` and having fallback `predict`